api_client/python/timesketch_api_client/client.py

```diff
@@ -368,23 +368,22 @@ def fetch_resource_data(self, resource_uri, params=None):
             Dictionary with the response data.
 
         Raises:
-            RuntimeError: If response could not be JSON-decoded after
-                DEFAULT_RETRY_COUNT attempts.
+            RuntimeError: If response cannot be interpreted or is empty.
         """
         resource_url = "{0:s}/{1:s}".format(self.api_root, resource_uri)
         response = self.session.get(resource_url, params=params)
 
-        retry_count = 0
-        while True:
-            result = error.get_response_json(response, logger)
-            # Any dict with content is good enough for us to return.
-            if result:
-                return result
-            retry_count += 1
-            if retry_count >= self.DEFAULT_RETRY_COUNT:
-                raise RuntimeError(
-                    f"Unable to fetch JSON resource data. Response: {str(result)}"
-                )
+        result = error.get_response_json(response, logger, url=resource_url)
```

Comment:
Why are you removing the retry here? It was added because sometimes the resource is not ready and a retry is needed. This will reintroduce issues that was fixed.

Reply:
I guess you want to delegate this to the client instead? I would argue that having a retry server side makes sense as well. With this we will need to change client code i several places in order to avoid subtle issues.

Reply:
> Why are you removing the retry here? It was added because sometimes the resource is not ready and a retry is needed. This will reintroduce issues that was fixed.

Yes, the idea was to delegate this to the client. However, I can see why we might want to have this also in the API client side covered.
Side note: The retry logic added here in #2552 does only re-try to parse the same response and does not do a new request to the server to my understanding.

I'll add a commit to fix both.